### PR TITLE
Circuit breaker

### PR DESCRIPTION
**Pull request by jairhenrique**

### CODE REVIEW

File: docs/failures.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Failures
```
Comment (renanivo) [inline docs/failures.md]: What about naming this module `fallbacks` instead of `failures`?

**Review by renanivo — APPROVED** [review]

File: docs/fallbacks.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Cache key where the number of errors is incremented.
+
+`max_failures`
+
+Maximum number of errors.
+
+`max_failure_exception`
+
+Exception who is raised when exceded maximum number of errors and when circuit is open.
```
Comment (RomuloOliveira) [inline docs/fallbacks.md]:
``` diff
- who
+ which
```

or

```
Exception raised when...
```

File: docs/fallbacks.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+`max_failure_timeout`
+
+This value is setted on first error. It is used to validate number of errors by time.
```
Comment (RomuloOliveira) [inline docs/fallbacks.md]:
``` diff
- setted
+ set
```

**Review by RomuloOliveira — APPROVED** [review]
👏

File: docs/fallbacks.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+Exception raised when exceded maximum number of errors and when circuit is open.
+
+`max_failure_timeout`
+
+This value is set on first error. It is used to validate number of errors by time.
+
+`circuit_timeout`
+
+Time of the circuit will be open.
```
Comment (jarussi-luizalabs) [inline docs/fallbacks.md]: Maybe `Time that the circuit will be open`  is better?

File: docs/fallbacks.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+`catch_exceptions`
+
+List of exception who is catched to increase number of errors.
```
Comment (jarussi-luizalabs) [inline docs/fallbacks.md]: Maybe `List of exception that is catched to incrase number of errors` is better?

**Review by jarussi-luizalabs — APPROVED** [review]

**Pull request merged**